### PR TITLE
48 printed message for add account should not be divided by 100

### LINKED ISSUE #48: Printed message for add account should not be divided by 100
image
As seen in the picture above, the balance in the account is correct, but the printed message in add account is wrong.

### PR DESCRIPTION
**Pull request by Bawfen**
# Description

## Type of Change

- [ ] Possible breaking change (Change in feature that may cause existing features to not work correctly, e.g. change in
  schema)
- [x] Non breaking change (Bug fix, New feature)

## What type of PR is this?

- [x] bug fix
- [x] cleanup
- [ ] documentation
- [ ] feature

## What this PR does / Why we need it:

```docs
Fixes bug with the display message for AddCommand, and also fixes a logic issue where the precision of adds were lost.
```

## Which issue(s) this PR fixes, if:

Fixes #48 

#### Additional documentation e.g., usage docs, testing docs etc.:

```docs

```

## Checklist

- [x] Changes have been tested
- [x] All automated tests pass
- [x] Relevant Tags and Milestones have been added

### CODE REVIEW

File: src/test/java/seedu/duke/commands/BalanceCommandTest.java [hunk 1]
```diff
@@ -89,37 +89,36 @@ public void getBalance_ifCurrencyIsNotSpecified_shouldReturnAllCurrencies() {
...
-                }
-                else if(account.getCurrencyType().equals(Currency.CNY)){
-                    CNYAccount = account;
+            Account cnyAccount = null;
+            Account eurAccount = null;
+            for (Account account : output) {
+                if (account.getCurrencyType().equals(Currency.EUR)) {
+                    eurAccount = account;
+                } else if (account.getCurrencyType().equals(Currency.CNY)) {
+                    cnyAccount = account;
```
Comment (woowenjun99) [inline src/test/java/seedu/duke/commands/BalanceCommandTest.java:98]: A bit weird here. Why run this in a loop? Why not just get the account directly and assert that it is equal to the currency?

File: src/test/java/seedu/duke/commands/BalanceCommandTest.java [hunk 2]
```diff
@@ -89,37 +89,36 @@ public void getBalance_ifCurrencyIsNotSpecified_shouldReturnAllCurrencies() {
...
                 }
             }
-            if(CNYAccount == null || EURAccount == null){
+            if (cnyAccount == null || eurAccount == null) {
                 fail();
             }
-            System.out.println(CNYAccount.getBalance());
-            System.out.println(EURAccount.getBalance());
+            System.out.println(cnyAccount.getBalance());
+            System.out.println(eurAccount.getBalance());
```
Comment (woowenjun99) [inline src/test/java/seedu/duke/commands/BalanceCommandTest.java:105]: Do we need the `System.out.println` in the test? If we are testing what is printed on the screen, we can follow what Arif has done on the other screen?

File: src/main/java/seedu/duke/commands/AddCommand.java [hunk 3]
```diff
@@ -70,6 +74,11 @@ public void execute(Ui ui, AccountList account) {
             ui.printMessage(ErrorMessage.INVALID_AMOUNT_TO_ADD);
         } catch (NullPointerException e) {
             ui.printMessage(ErrorMessage.NO_AMOUNT_PROVIDED);
+        } catch (NotEnoughInAccountException e) {
+            // this should not happen since we are adding money
+            ui.printMessage(ErrorMessage.NOT_ENOUGH_IN_ACCOUNT);
+        } catch (InvalidUpdateBalanceActionException e) {
```
Comment (woowenjun99) [inline src/main/java/seedu/duke/commands/AddCommand.java:80]: Do you want to remove this error?

**Review by woowenjun99 — COMMENTED** [review]
I think overall ok, I just feel that:
- [ ] The formatting should be done in another branch or issue because this makes it difficult to perform code review, unless the formatting change is a result of the formatter when we do code review. The same goes for future bugs or issues, I don't think we should try fixing everything in one branch unless it affects the behaviour
- [ ] The code LGTM, I think it is ok to merge. I am just confused about why the tests were implemented in a certain manner.
- [ ] I think all our formatters are set differently, My formatter does not have a space between the parentheses and the function name. Maybe someone can do the formatting for all files

File: src/main/java/seedu/duke/commands/AddCommand.java (hunk 3 above)
Comment (Bawfen) [reply src/main/java/seedu/duke/commands/AddCommand.java:80]: It is thrown by the function so unless we separate into different functions or the function prints on its own it cannot be helped

File: src/test/java/seedu/duke/commands/BalanceCommandTest.java (hunk 2 above)
Comment (Bawfen) [reply src/test/java/seedu/duke/commands/BalanceCommandTest.java:105]: I didn't want to change the logic, @Arif-Khalid can you comment on this

Comment (woowenjun99) [reply src/test/java/seedu/duke/commands/BalanceCommandTest.java:105]: Noted, closing this

**Comment by woowenjun99** [comment]
LGTM

**Pull request merged**